src/Workspaces/CSharp/Portable/Extensions/CastExpressionSyntaxExtensions.cs

```diff
@@ -540,5 +555,22 @@ private static bool IsRequiredImplicitNumericConversion(ITypeSymbol sourceType,
                     return false;
             }
         }
+
+        private static bool IsInUncheckedContext(CastExpressionSyntax cast)
+        {
+            SyntaxNode currentNode = cast;
+
+            do
+            {
+                if (currentNode.Kind() == SyntaxKind.UncheckedExpression || currentNode.Kind() == SyntaxKind.UncheckedStatement)
+                {
+                    return true;
+                }
+
+                currentNode = currentNode.Parent;
+            } while (currentNode is ExpressionSyntax || currentNode is StatementSyntax);
```

Comment:
I agree for unchecked statements, but what if the lambda is in an unchecked expression?  Also, should you stop if you encounter a checked statement/expression?